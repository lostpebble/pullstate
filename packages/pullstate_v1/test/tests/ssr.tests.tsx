import React from "react";
import {
  createPullstateCore,
  InjectStoreState, IPullstateInstanceConsumable,
  PullstateProvider,
  Store,
  update,
  useStoreState,
} from "../../src/index";
import ReactDOMServer from "react-dom/server";
import { ITestUIStore, TestUIStore } from "./testStores/TestUIStore";
const beautify = require('js-beautify').html;

const PullstateCore = createPullstateCore({ TestUIStore });

const Counter = () => {
  const { TestUIStore: ui } = PullstateCore.useStores();
  const count = useStoreState(ui, s => s.count);

  return (
    <div>
      <b>{count}</b> -{" "}
      <button
        onClick={() =>
          update(ui, s => {
            s.count++;
          })
        }
      >
        +
      </button>
    </div>
  );
};

const App = () => {
  const { TestUIStore: ui } = PullstateCore.useStores();

  return (
    <div>
      <h1>Some test</h1>
      <InjectStoreState store={ui} on={s => s.message}>
        {message => (
          <div>
            <h2>{message}</h2>
            <input
              onChange={e =>
                update(ui, s => {
                  s.message = e.target.value;
                })
              }
              value={message}
            />
          </div>
        )}
      </InjectStoreState>
      <InjectStoreState store={ui}>{uiStore => <h2>{uiStore.count}</h2>}</InjectStoreState>
      <Counter />
    </div>
  );
};

describe("Server Side Rendering tests", () => {
  const instance = PullstateCore.instantiate({ ssr: true });

  instance.stores.TestUIStore.update(s => {
    s.message = "hey there!";
  });

  const ReactApp = (
    <PullstateProvider instance={instance}>
      <App />
    </PullstateProvider>
  );

  it("Should be able to display data that's been changed on the server directly", () => {
    expect(beautify(ReactDOMServer.renderToString(ReactApp))).toMatchSnapshot();
  });
});
