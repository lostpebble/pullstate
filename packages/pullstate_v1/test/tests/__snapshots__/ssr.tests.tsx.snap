// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server Side Rendering tests Should be able to display data that's been changed on the server directly 1`] = `
"<div>
    <h1>Some test</h1>
    <div>
        <h2>hey there!</h2><input value=\\"hey there!\\" />
    </div>
    <h2>5</h2>
    <div><b>5</b> -
        <!-- --> <button>+</button></div>
</div>"
`;
